packages/js/components/src/experimental-tree-control/hooks/use-keyboard.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import { LinkedTree } from '../types';
+
+function getFirstChild(
+	currentHeading: HTMLDivElement
+): HTMLLabelElement | null {
+	const parentTreeItem = currentHeading?.closest< HTMLDivElement >(
+		'.experimental-woocommerce-tree-item'
+	);
+	const firstSubTreeItem = parentTreeItem?.querySelector(
+		'.experimental-woocommerce-tree > .experimental-woocommerce-tree-item'
+	);
+	const label = firstSubTreeItem?.querySelector< HTMLLabelElement >(
+		'.experimental-woocommerce-tree-item__heading > .experimental-woocommerce-tree-item__label'
+	);
+	return label ?? null;
+}
+
+function getFirstAncestor(
+	currentHeading: HTMLDivElement
+): HTMLLabelElement | null {
+	const parentTree = currentHeading?.closest< HTMLDivElement >(
+		'.experimental-woocommerce-tree'
+	);
+	const grandParentTreeItem = parentTree?.closest< HTMLDivElement >(
+		'.experimental-woocommerce-tree-item'
+	);
+	const label = grandParentTreeItem?.querySelector< HTMLLabelElement >(
+		'.experimental-woocommerce-tree-item__heading > .experimental-woocommerce-tree-item__label'
+	);
+	return label ?? null;
+}
+
+function getAllHeadings(
+	currentHeading: HTMLDivElement
+): NodeListOf< HTMLDivElement > | undefined {
+	const rootTree = currentHeading.closest< HTMLDivElement >(
+		'.experimental-woocommerce-tree--level-1'
+	);
+	return rootTree?.querySelectorAll< HTMLDivElement >(
+		'.experimental-woocommerce-tree-item > .experimental-woocommerce-tree-item__heading'
+	);
+}
+
+function getNextFocusableElement(
+	currentHeading: HTMLDivElement,
+	code: 'ArrowDown' | 'ArrowUp'
+): HTMLLabelElement | null {
+	const headingsNodeList = getAllHeadings( currentHeading );
+	if ( ! headingsNodeList ) return null;
+
+	let currentHeadingIndex = 0;
+	for ( const heading of headingsNodeList.values() ) {
+		if ( heading === currentHeading ) break;
+		currentHeadingIndex++;
+	}
+	if (
+		currentHeadingIndex < 0 ||
+		currentHeadingIndex >= headingsNodeList.length
+	) {
+		return null;
+	}
+
+	const step = {
```

Comment:
I think you can move this to outside the function as it's a constant

Reply:
I put it there because I intentionally wanted to hide the `step` variable from being used outside of that function.

But I don't have a strong opinion for this anyway. It's done now!